heat/core/tests/test_suites/basic_test.py

```diff
@@ -0,0 +1,173 @@
+
+    def assert_func_equal(self, tensor, heat_func, numpy_func, distributed_result=True, heat_args=None, numpy_args=None):
+        """
+        Checks if a heat function works equivalent to a given numpy function. The initial array is iteratively split
+        along each axis before the heat function is applied. After that the resulting arrays are compared using the
+        assert_array_equal function.
+
+        Parameters
+        ----------
+        tensor: tuple, list, numpy.ndarray or torch.Tensor
+            The tensor to check the function against.
+            If a tuple or list is provided instead a random array of the specified shape is generated
+        heat_func: function
+            The function that is to be tested
+        numpy_func: function
+            The numpy implementation of an equivalent function to test against
+        distributed_result: bool, optional
+            Specify whether the result of the heat function is distributed across all nodes or all nodes have the full
+            result. Default is True.
+        heat_args: dictionary, optional
+            The keyword arguments that will be passed to the heat function. Array and split function don't need to be
+            specified. Default is {}.
+        numpy_args:
+            The keyword arguments that will be passed to the numpy function. Array doesn't need to be specified.
+            Default is {}.
+
+        Raises
+        ------
+        AssertionError if the functions to not perform equally.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import heat as ht
+        >>> a = np.arange(10)
+        >>> self.assert_func_equal(a, ht.exp, np.exp)
+
+        >>> self.assert_func_equal(a, ht.exp, np.log)
+        AssertionError: [...]
+        >>> self.assert_func_equal((1, 3, 5), ht.any, np.any, distributed_result=False)
+
+        >>> heat_args = {'sorted': True, 'axis': 0}
+        >>> numpy_args = {'axis': 0}
+        >>> self.assert_func_equal([5, 5, 5, 5], ht.unique, np.unique, heat_arg=heat_args, numpy_args=numpy_args)
+        """
+        self.assertTrue(callable(heat_func))
+        self.assertTrue(callable(numpy_func))
+        if isinstance(tensor, tuple) or isinstance(tensor, list):
+            tensor = self._create_random_array(tensor)
+
+        if heat_args is None:
+            heat_args = {}
+        if numpy_args is None:
+            numpy_args = {}
+
+        if isinstance(tensor, np.ndarray):
+            torch_tensor = torch.from_numpy(tensor.copy())
+            np_array = tensor
+        elif isinstance(tensor, torch.Tensor):
+            torch_tensor = tensor
+            np_array = tensor.numpy().copy()
+        else:
+            raise TypeError('The input tensors type must be one of [tuple, list, ' +
+                            'numpy.ndarray, torch.tensor] but is {}'.format(type(tensor)))
+
+        np_res = numpy_func(np_array, **numpy_args)
```

Comment:
dtype (float64 vs float32)

Reply:
The user is always able to define this using the `numpy_args`. I want to keep the function general and therefore not make too many assumptions about what the user actually wants.